changes/18034-builtin-labels

```diff
@@ -0,0 +1 @@
+- Updated label endpoints and UI to prevent creating, updating, or deleting built-in labels.
```

Comment:
UI changes improve label-related error messages, typing, and resolves a page crash that otherwise occurs when a user attempts to directly navigate to the manage hosts page using the label id of the "Ubuntu Linux" or "Red Hat Linux" built-in labels, which happened because the platform icon map was being accessed indiscriminately and label dropdown was not properly handling the case where hidden platform drop down option doesn't have an associated icon.